File: src/main/kotlin/com/huawei/cangjie/psi/CjEnumEntry.java
package com.huawei.cangjie.psi;

import com.huawei.cangjie.psi.stubs.CangJiePlaceHolderStub;
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes;
import com.intellij.lang.ASTNode;
import com.intellij.psi.stubs.IStubElementType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public  class CjEnumEntry  extends CjElementImplStub<CangJiePlaceHolderStub<? extends CjEnumEntry>>{


    public CjEnumEntry(@NotNull ASTNode node) {
        super(node);
    }

    public CjEnumEntry(
            @NotNull CangJiePlaceHolderStub<? extends CjEnumEntry> stub,
            @NotNull IStubElementType nodeType) {
        super(stub, nodeType);
    }

    @Override
    public <R, D> R accept(@NotNull CjVisitor<R, D> visitor, D data) {
        return visitor.visitEnumEntry(this, data);
    }
    @Override
    public String toString() {
        return   getNode().getElementType().toString();
    }
    @Nullable
    public CjTypeReference getTypeReference() {
        return getStubOrPsiChild(CjStubElementTypes.TYPE_REFERENCE);
    }

}

File: src/main/kotlin/com/huawei/cangjie/psi/CjEnumEntry.kt
package com.huawei.cangjie.psi

import com.huawei.cangjie.psi.stubs.CangJieEnumEntryStub
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes
import com.intellij.lang.ASTNode
import com.intellij.psi.stubs.IStubElementType

class CjEnumEntry : CjTypeStatement {
    constructor(node: ASTNode) : super(node)

    constructor(
        stub: CangJieEnumEntryStub

    ) : super(stub,  CjStubElementTypes.ENUM_ENTRY)

    override fun <R, D> accept(visitor: CjVisitor<R, D>, data: D?): R {
        return visitor.visitEnumEntry(this, data)
    }

    override fun toString(): String {
        return node.elementType.toString()
    }

    val typeReference: CjTypeReference?
        get() = getStubOrPsiChild(CjStubElementTypes.TYPE_REFERENCE)
}
